src/plugins/external_tool_support/src/ExternalToolManager.cpp

```diff
@@ -193,28 +190,42 @@ QString ExternalToolManagerImpl::addTool(ExternalTool *tool) {
         foreach (const QString &dependency, toolDependencies) {
             dependencies.insertMulti(dependency, tool->getId());
         }
+    }
+}
+
+void ExternalToolManagerImpl::addToolToListsIfNecessary(const ExternalTool &tool) {
+    if (!tool.isValid()) {
+        if (tool.getDependencies().isEmpty() || dependenciesAreOk(tool.getId())) {
+            validateList << tool.getId();
+        }
+    }
+
+    if (!validateList.contains(tool.getId()) && !tool.isModule() && !tool.isValid()) {
+        searchList << tool.getId();
+    }
+}
 
-        if (dependenciesAreOk(tool->getId()) && !tool->isValid()) {
-            if (tool->isModule()) {
-                QString masterId = tool->getDependencies().first();
+QString ExternalToolManagerImpl::findToolPath(const ExternalTool &tool) {
+    QString toolPath;
+    SAFE_POINT(nullptr != etRegistry, "The external tool registry is nullptr", toolPath);
+
+    QStringList toolDependencies = tool.getDependencies();
+    if (!toolDependencies.isEmpty()) {
+        if (dependenciesAreOk(tool.getId()) && !tool.isValid()) {
+            if (tool.isModule()) {
+                QString masterId = tool.getDependencies().first();
                 ExternalTool *masterTool = etRegistry->getById(masterId);
-                SAFE_POINT(masterTool, QString("An external tool '%1' isn't found in the registry").arg(masterId), "");
+                SAFE_POINT(nullptr != masterTool, QString("An external tool '%1' isn't found in the registry").arg(masterId), toolPath);
 
                 toolPath = masterTool->getPath();
             } else {
-                toolPath = tool->getPath();
+                toolPath = tool.getPath();
             }
-            validateList << tool->getId();
+        } else {
         }
+    } else if (!tool.isValid()) {
+        toolPath = tool.getPath();
     } else {
```

Comment:
1 more empty else.
If you want to express something with a code like this please add a comment